lib/cc/services/github_pull_requests.rb

```diff
@@ -13,13 +14,53 @@ class Config < CC::Service::Config
       label: "Github Context",
       description: "The integration name next to the pull request status",
       default: "codeclimate"
+    attribute :welcome_comment_enabled, Axiom::Types::Boolean,
+      label: "Welcome comment enabled?",
+      description: "Post a welcome comment?",
+      default: false
+    attribute :welcome_comment_markdown, Axiom::Types::String,
+      label: "Welcome comment markdown",
+      description: "The body of the welcome comment for first-time contributors to this repo.",
+      default: CC::Service::GitHubPullRequestsWelcomeCommentPresenter::DEFAULT_BODY
 
     validates :oauth_token, presence: true
   end
 
   self.title = "GitHub Pull Requests"
   self.description = "Update pull requests on GitHub"
 
+  CANT_POST_COMMENTS_MESSAGE = "Access token is invalid - can't post comments".freeze
+  INVALID_TOKEN_MESSAGE = "Access token is invalid.".freeze
+
+  MESSAGES = {
+    [true, true] => VALID_TOKEN_MESSAGE,
+    [true, nil] => VALID_TOKEN_MESSAGE,
+    [false, nil] => CANT_UPDATE_STATUS_MESSAGE,
+    [true, false] => CANT_POST_COMMENTS_MESSAGE,
+    [false, true] => CANT_UPDATE_STATUS_MESSAGE,
+    [false, false] => INVALID_TOKEN_MESSAGE,
+  }.freeze
+
+  # Override:
+  def receive_test
+    setup_http
```

Comment:
Would something like this work, leveraging the default behavior which tests token validity?

```ruby
def receive_test
  super

  if able_to_post_comments?
    { ok: true, message: "Access token is valid." }
  else
    { ok: false, message: "Access token does not have full repo scope." }
  end
end
```